api/src/devcomp/domain/models/Module.js

```diff
@@ -2,38 +2,28 @@ import { assertNotNullOrUndefined } from '../../../shared/domain/models/asserts.
 import { NotFoundError } from '../../../shared/domain/errors.js';
 
 class Module {
-  #id;
-  #slug;
-  #title;
-  #grains;
-
-  constructor({ id, slug, title, grains }) {
+  constructor({ id, slug, title, grains, transitionTexts = [] }) {
     assertNotNullOrUndefined(id, "L'id est obligatoire pour un module");
     assertNotNullOrUndefined(title, 'Le titre est obligatoire pour un module');
     assertNotNullOrUndefined(slug, 'Le slug est obligatoire pour un module');
     assertNotNullOrUndefined(grains, 'Une liste de grains est obligatoire pour un module');
     this.#assertGrainsIsAnArray(grains);
+    this.#assertTransitionTextsLinkedToGrain(transitionTexts, grains);
 
-    this.#id = id;
-    this.#slug = slug;
-    this.#title = title;
-    this.#grains = grains;
-  }
-
-  get id() {
-    return this.#id;
-  }
-
-  get slug() {
-    return this.#slug;
-  }
-
-  get title() {
-    return this.#title;
+    this.id = id;
+    this.slug = slug;
+    this.title = title;
+    this.grains = grains;
+    this.transitionTexts = transitionTexts;
   }
 
-  get grains() {
-    return this.#grains;
+  #assertTransitionTextsLinkedToGrain(transitionTexts, grains) {
+    const assertTransitionTextsLinkedToGrain = transitionTexts.every(
```

Comment:
nitpick [naming]: je pense que c'est plus claire de pas reutiliser le meme nom pour le constant et la function
```suggestion
    const isTransitionTextsLinkedToGrain = transitionTexts.every(
```